screen/contributors/src/main/java/com/ivy/contributors/IvyWalletRepositoryDataSource.kt

```diff
@@ -33,27 +39,59 @@ class IvyWalletRepositoryDataSource @Inject constructor(
         @SerialName("stargazers_count")
         val stars: Int,
         @SerialName("html_url")
-        val url: String
+        val url: String,
     )
 
     companion object {
         private const val CONTRIBUTORS_PER_PAGE = 100
+        private const val DISPLAY_ANONYMOUS_CONTRIBUTORS = true
+        private const val INITIAL_PAGE = 1
     }
 
-    suspend fun fetchContributors(): List<ContributorDto>? {
-        return try {
-            withContext(Dispatchers.IO) {
-                httpClient
-                    .get("https://api.github.com/repos/Ivy-Apps/ivy-wallet/contributors") {
-                        parameter("per_page", CONTRIBUTORS_PER_PAGE)
-                    }
-                    .body<List<ContributorDto>>()
+    suspend fun fetchContributors(): Either<String, List<ContributorDto>> = either {
```

Comment:
This `either` isn't needed

Reply:
We should use either {} to use the catch function and make the Either type explicit to avoid getting an `Any` type."

Reply:
Because in `catch` the last line of the lambdas must return `Either` - see other catch usages and the function signature

Reply:
`catch({ Either.Right(42) }){ Either.Left("err" ) }` you can use it like that @rodrigoliveirac

Reply:
Ok. Thanks!